fs2/src/main/scala/io/finch/fs2/package.scala

```diff
@@ -68,12 +70,11 @@ trait StreamInstances {
         .compile
         .drain
 
-      F.toIO(run).unsafeRunAsyncAndForget()
-      p
+      F.productR(F.runAsync(run)(this).to[F])(F.pure(p))
```

Comment:
@joroKr21 I took your suggestion.

Reply:
Wait, what does it mean to pass `this` as the callback?